primitives/core/Cargo.toml

```diff
@@ -48,7 +48,7 @@ bitflags = "1.3"
 
 # full crypto
 ed25519-zebra = { version = "3.0.0", default-features = false, optional = true}
-blake2-rfc = { version = "0.2.18", default-features = false, optional = true }
+blake2 = { version = "0.10.2", default-features = false, optional = true }
```

Comment:
1. Hmm... this crate has also some [simd-related features](https://github.com/RustCrypto/hashes/blob/master/blake2/Cargo.toml#L25); could you try turning them on and see if it makes a difference performance-wise? If so we might consider turning them on by default (but only when `std` is enabled).
2. It looks like the newest version is 0.10.4; maybe you could upgrade it while you're at it?

Reply:
Currently the simd features cannot be used on stable channel.
In particular require the `repr_simd` feature.

Nevertheless I've tried with nightly and surprisingly using `simd`
instructions results in slower performances.

### 4K blocks

plain       time:   [11.000 µs 11.003 µs 11.005 µs]
simd-asm    time:   [13.606 µs 13.607 µs 13.608 µs]
simd        time:   [14.049 µs 14.053 µs 14.057 µs]

### 4MB blocks

plain:      time:   [10.742 ms 10.744 ms 10.746 ms]
simd-asm    time:   [13.383 ms 13.386 ms 13.389 ms]
simd        time:   [13.786 ms 13.787 ms 13.789 ms]


---

I will updated `blake2` to 0.10.4

Reply:
Okay, that's interesting; they must have either done something very wrong, or perhaps the plain case is autovectorized and their SIMD code's just suboptimal.

Anyway, thanks for checking this out!

Reply:
FYI I reported simd concerns here upstream 2 days ago: https://github.com/RustCrypto/hashes/issues/407

Reply:
@nazar-pc just for reference:
- https://github.com/RustCrypto/hashes/issues/360#issuecomment-1248132103
- https://github.com/paritytech/parity-db/pull/111